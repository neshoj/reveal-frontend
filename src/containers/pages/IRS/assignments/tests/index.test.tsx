import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import { IRSAssignmentPlansList } from '../';
import * as fixtures from '../../../../../store/ducks/generic/tests/fixtures';
import { PlanRecord } from '../../../../../store/ducks/plans';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

describe('components/IRS Reports/IRSPlansList', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const props = {
      plans: [],
    };
    shallow(
      <Router history={history}>
        <IRSAssignmentPlansList {...props} />
      </Router>
    );
  });

  it('renders plan definition list correctly', () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const plans = fixtures.planRecords as PlanRecord[];
    const props = {
      plans,
    };
    const wrapper = mount(
      <Router history={history}>
        <IRSAssignmentPlansList {...props} />
      </Router>
    );
    expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');
    expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('page title');
    expect(toJson(wrapper.find('thead tr th'))).toMatchSnapshot('table headers');
    expect(toJson(wrapper.find('tbody tr td'))).toMatchSnapshot('table rows');
    wrapper.unmount();
  });
});
