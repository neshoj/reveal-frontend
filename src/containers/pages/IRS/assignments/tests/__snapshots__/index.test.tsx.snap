// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: breadcrumbs 1`] = `
Array [
  <li
    className="breadcrumb-item"
  >
    <Link
      key="0"
      replace={false}
      to="/"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </Link>
  </li>,
  <li
    aria-current="page"
    className="active breadcrumb-item"
  >
    Assign Plans
  </li>,
]
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: page title 1`] = `
<h3
  className="mt-3 mb-3 page-title"
>
  Assign Plans
</h3>
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table headers 1`] = `
Array [
  <th
    className="listview-td"
    key="0"
  >
    Title
  </th>,
  <th
    className="listview-td"
    key="1"
  >
    Intervention
  </th>,
  <th
    className="listview-td"
    key="2"
  >
    Start Date
  </th>,
  <th
    className="listview-td"
    key="3"
  >
    End Date
  </th>,
  <th
    className="listview-td"
    key="4"
  >
    Plan Status
  </th>,
]
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table rows 1`] = `
Array [
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="727c3d40-e118-564a-b231-aac633e6abce"
      replace={false}
      to="/assign/727c3d40-e118-564a-b231-aac633e6abce"
    >
      <a
        href="/assign/727c3d40-e118-564a-b231-aac633e6abce"
        onClick={[Function]}
      >
        IRS 2019-09-05 TEST
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    IRS
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-09-05
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-25
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    retired
  </td>,
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="02e358ed-87fe-5720-8782-0afe28cb7e8c"
      replace={false}
      to="/assign/02e358ed-87fe-5720-8782-0afe28cb7e8c"
    >
      <a
        href="/assign/02e358ed-87fe-5720-8782-0afe28cb7e8c"
        onClick={[Function]}
      >
        IRS 2019-08-30m
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    IRS
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-08-30
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-14
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    active
  </td>,
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
      replace={false}
      to="/assign/c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
    >
      <a
        href="/assign/c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
        onClick={[Function]}
      >
        Berg Namibia 2019
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    IRS
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-08-29
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-18
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    active
  </td>,
]
`;
